src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1447,6 +1454,149 @@ void Lowering::LowerHWIntrinsicCmpOp(GenTreeHWIntrinsic* node, genTreeOps cmpOp)
+    NamedIntrinsic blendVariableId;
+
+    if (simdSize == 32) 
+    {
+        // for Vector256, BlendVariable for floats is available on AVX, whereas other types require AVX2
+        if (varTypeIsFloating(simdBaseType)) 
+        {
+            // This should have already been confirmed
+            assert(comp->compIsaSupportedDebugOnly(InstructionSet_AVX));
+            supportsBlendVariable = true;
+            blendVariableId = NI_AVX_BlendVariable;
+        }
+        else 
+        {
+            supportsBlendVariable = comp->compOpportunisticallyDependsOn(InstructionSet_AVX2);
+            blendVariableId = NI_AVX2_BlendVariable;
+        }
+    } 
+    else 
+    {
+        // for Vector128, BlendVariable is available on SSE41
+        supportsBlendVariable = comp->compOpportunisticallyDependsOn(InstructionSet_SSE41);
+        blendVariableId = NI_SSE41_BlendVariable;
+    }
+
+    if (op1->OperIsHWIntrinsic() && supportsBlendVariable)
+    {
+        GenTreeHWIntrinsic* hwIntrinsic = op1->AsHWIntrinsic();
+        NamedIntrinsic id = hwIntrinsic->GetHWIntrinsicId();
+
+        // If the condition is a per-element mask, we can optimize
+        if (HWIntrinsicInfo::ReturnsPerElementMask(id)) 
+        {
+            // result = BlendVariable op2 op3 op1
+            node->ResetHWIntrinsicId(blendVariableId, comp, op2, op3, op1);
+            return;
+        }
+    }
+
+    // We cannot optimize, so produce unoptimized instructions
+
+    // We will be constructing the following parts:
+    //          /--*  op1 simd16
+    //          *  STORE_LCL_VAR simd16
+    //   op1  =    LCL_VAR       simd16
+    //   tmp1 =    LCL_VAR       simd16
+    //   ...
+
+    GenTree* tmp1;
+    GenTree* tmp2;
+    GenTree* tmp3;
+
+    LIR::Use op1Use(BlockRange(), &node->Op(1), node);
+    ReplaceWithLclVar(op1Use);
+    op1 = node->Op(1);
+
+    tmp1 = comp->gtClone(op1);
+    BlockRange().InsertAfter(op1, tmp1);
+
+    // ...
+    // tmp2 = op1 & op2
+    // ...
+    tmp2 = comp->gtNewSimdBinOpNode(GT_AND, simdType, op1, op2, simdBaseJitType, simdSize, isSimdAsHWIntrinsic);
+    BlockRange().InsertAfter(op2, tmp2);
+    LowerNode(tmp2);
+
+    // ...
+    // tmp3 = op3 & ~tmp1
+    // ...
+    tmp3 = comp->gtNewSimdBinOpNode(GT_AND_NOT, simdType, op3, tmp1, simdBaseJitType, simdSize, isSimdAsHWIntrinsic);
+    BlockRange().InsertAfter(op3, tmp3);
+    LowerNode(tmp3);
+
+    // determine which Or intrinsic to use, depending on target architecture
+    NamedIntrinsic orIntrinsic = NI_Illegal;
+    
+    if (simdSize == 32)
+    {
+        assert(comp->compIsaSupportedDebugOnly(InstructionSet_AVX));
+
+        if (varTypeIsFloating(simdBaseType))
+        {
+            orIntrinsic = NI_AVX_Or;
+        }
+        else if (comp->compOpportunisticallyDependsOn(InstructionSet_AVX2))
+        {
+            orIntrinsic = NI_AVX2_Or;
+        }
+        else
+        {
+            // Since this is a bitwise operation, we can still support it by lying
```

Comment:
maybe this is obvious to jit devs, but I expected to see a comment explaining the "still" here. Eg., "we can't do foo because bar ... we can still support it by lying. .."
or maybe just explain why you have to lie.

Reply:
ah, I guess it simply means, not of the previous two blocks.